### DISCUSSION TITLE
[Feature] Load default config file from classpath with Enviroment variable override

### DISCUSSION THREAD

**Discussion by alievrenkut** [apache/skywalking#12772] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Load default config file from classpath so that we can override it with environment variables.
For example it is not possible to start agent while we already have SW_AGENT_NAME env property. 
This only works if we put config file next to jar file. It should work without any config file.

### Use case

I want to only write `-javaagent:/path/to/skywalking-agent.jar` and agent picks configuration from environment variables.

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
We don't have a plan to support this. Mostly, people is using container and k8s to deploy already.
Can't see why you can't have system env variable setup, using -D, or provide a new config file in a folder. That is an easy deployment setup.

**Comment by alievrenkut** [comment]
Sorry, My mistake I was thinking only skywalking-agent.jar is needed to work but I see that we need other folders like plugins, activations, ... etc

I was thinking so that we supply single jar but now I see it will not be that easy.
